controlplane/kubeadm/internal/workload_cluster_coredns_test.go

```diff
@@ -286,63 +290,77 @@ kind: ClusterConfiguration
 	// We are using testEnv as a workload cluster, and given that each test case assumes well known objects with specific
 	// Namespace/Name (e.g. The CoderDNS ConfigMap & Deployment, the kubeadm ConfigMap), it is not possible to run the use cases in parallel.
 	for _, tt := range tests {
-		g := NewWithT(t)
-		t.Log(tt.name)
-
-		for _, o := range tt.objs {
-			// NB. deep copy test object so changes applied during a test does not affect other tests.
-			o := o.DeepCopyObject().(client.Object)
-			g.Expect(testEnv.Create(ctx, o)).To(Succeed())
-		}
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
 
-		w := &Workload{
-			Client:          testEnv.GetClient(),
-			CoreDNSMigrator: tt.migrator,
-		}
-		err := w.UpdateCoreDNS(ctx, tt.kcp)
-		if tt.expectErr {
-			g.Expect(err).To(HaveOccurred())
-			return
-		}
-		g.Expect(err).ToNot(HaveOccurred())
+			for _, o := range tt.objs {
+				// NB. deep copy test object so changes applied during a test does not affect other tests.
+				o := o.DeepCopyObject().(client.Object)
+				g.Expect(testEnv.Create(ctx, o)).To(Succeed())
+				// this makes sure that the cache is updated with the object
+				// to avoid 404 errors leading to test flakes
+				g.Eventually(func() bool {
+					key, _ := client.ObjectKeyFromObject(o)
+					err := testEnv.Get(ctx, key, o)
+					if err != nil && apierrors.IsNotFound(err) {
+						return false
+					}
+					return true
```

Comment:
I _think_ we might want this as
```go
if err == nil {
  return true
}
return false
```

Otherwise, if Get returns some error other than NotFound, we'd exit the eventually loop incorrectly

Reply:
+1

Reply:
Good catch, making the change.